src/AggregateFunctions/AggregateFunctionDeltaSum.h

```diff
@@ -0,0 +1,108 @@
+#pragma once
+
+#include <type_traits>
+#include <experimental/type_traits>
+
+#include <IO/ReadHelpers.h>
+#include <IO/WriteHelpers.h>
+
+#include <Columns/ColumnVector.h>
+#include <DataTypes/DataTypesDecimal.h>
+#include <DataTypes/DataTypesNumber.h>
+
+#include <AggregateFunctions/IAggregateFunction.h>
+
+
+namespace DB
+{
+template <typename T>
+struct AggregationFunctionDeltaSumData
+{
+    T sum = 0;
+    bool seen_last = false;
+    T last = 0;
+    bool seen_first = false;
+    T first = 0;
+};
```

Comment:
Consider to rearrange values in this structure like:
```
template <typename T>
struct AggregationFunctionDeltaSumData
{
     T sum = 0;
     T last = 0;
     T first = 0;
     bool seen_first = false;
     bool seen_last = false;
};
```
Example:

```
template <typename T>
struct First
{
     T sum = 0;
     bool seen_last = false;
     T last = 0;
     bool seen_first = false;
     T first = 0;
};

template <typename T>
struct Second
{
     T sum = 0;
     T last = 0;
     T first = 0;
     bool seen_last = false;
     bool seen_first = false;
};
```

If for example T will be uint64_t sizeof(First) == 40, but sizeof(Second) == 32